apps/src/componentLibrary/button/Button.tsx

```diff
@@ -1,36 +1,82 @@
-import React from 'react';
-import classNames from 'classnames';
+import React, {memo} from 'react';
 
 import {ComponentSizeXSToL} from '@cdo/apps/componentLibrary/common/types';
-import moduleStyles from './button.module.scss';
+import {FontAwesomeV6IconProps} from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+
+import _BaseButton from './_baseButton/_BaseButton';
+import {ButtonType, ButtonColor} from './types';
+
+export const buttonColors: {[key in ButtonColor]: ButtonColor} = {
+  purple: 'purple',
+  black: 'black',
+  white: 'white',
+};
 
 export interface ButtonProps {
+  /** Button Component type */
+  type?: ButtonType;
+  /** Custom class name */
+  className?: string;
+  /** Button id */
+  id?: string;
+  /** Button color */
+  color?: ButtonColor;
+  /** Button text */
+  text?: string;
+  /** Is button disabled */
+  disabled?: boolean;
+  /** Is button pending */
+  isPending?: boolean;
+  /** Button aria-label */
+  ariaLabel?: string;
   /** Size of button */
   size?: ComponentSizeXSToL;
+  /** Left Button icon */
+  iconLeft?: FontAwesomeV6IconProps;
+  /** Button icon (When used in IconOnly mode)*/
+  icon?: FontAwesomeV6IconProps;
+  /** Left Button icon */
+  iconRight?: FontAwesomeV6IconProps;
+  /** Whether we use \<a> (when set to true) or \<button> (when false) html tag for Button component.
+   * If we want button to redirect to another page or download some file we should use \<a> tag.
+   * If we want button to call some function or submit some form we should use \<button> tag.
+   * */
+  useAsLink?: boolean;
+  /** (\<button> specific prop)
+   * Button html element type */
+  buttonType?: 'submit' | 'button';
```

Comment:
nit: I feel like the word `type` is getting a bit overloaded in these props. Maybe `tagType` or `buttonTag` for more specificity?

Reply:
Not sure about this one since it specifies `type` attribute of `<button>` element as documented.
`buttonTagType` or even `buttonTagButtonType` might be selfdocumenting names, though I think they're kinda too much 😄 
`tagType` or `buttonTag` might lead to even more confusion I think since that's not a tag prop and not related to a link button